addons/account/models/account_move.py

```diff
@@ -409,6 +409,7 @@ def _sequence_year_range_monthly_regex(self):
         help="Fiscal positions are used to adapt taxes and accounts for particular "
              "customers or sales orders/invoices. The default value comes from the customer.",
     )
+    partner_country = fields.Char(string='Partner Country', related='partner_id.country_id.name')
```

Comment:
```suggestion
    partner_country = fields.Many2one(string='Partner Country', related='partner_id.country_id')
```
We can just group on a Many2one